.github/workflows/harja_pr_test_and_build.yml

```
@@ -179,11 +174,63 @@ jobs:
       - name: Setup clojure environment
         uses: ./.github/actions/setup-clojure-env
 
-      # TODO: Aja halutut testit
-      #- name: Run unit tests jne.
-      #  run: lein test2junit
+      - name: Run backend tests
+        run: lein test
+
+      - name: run slow tests
+        run: lein test :hidas
+
+      - name: Front end tests
+        env:
+          PHANTOMJS_VERSIO: 2.1.1
+        run: |
+          cd /tmp
+          wget -O ~/phantomjs-${PHANTOMJS_VERSIO}-linux-x86_64.tar.bz2 https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-${PHANTOMJS_VERSIO}-linux-x86_64.tar.bz2
+          if [ "$(sha512sum /home/runner/phantomjs-${PHANTOMJS_VERSIO}-linux-x86_64.tar.bz2 )" != "039b62ecf2fa4196357e129727871a53b046948e17b21900c78a1af4d0b523e41b9d4137e7454b0a638333d7fc27e65d14f4af8c665e982a504c12f828525419  /home/runner/phantomjs-${PHANTOMJS_VERSIO}-linux-x86_64.tar.bz2" ]; then exit 1; fi;
+          tar -xf /home/runner/phantomjs-${PHANTOMJS_VERSIO}-linux-x86_64.tar.bz2 -C /tmp/
+          mv -f /tmp/phantomjs-${PHANTOMJS_VERSIO}-linux-x86_64/bin/phantomjs /usr/local/bin/phantomjs
+          rm /home/runner/phantomjs-${PHANTOMJS_VERSIO}-linux-x86_64.tar.bz2
+          cd /home/runner/work/harja/harja/
+          lein with-profile +test doo phantom test once
+
+  # integraatiotestit vaativat ActiveMQ-brokerin, siksi oma jobi
+  integration-tests:
+    name: "integraatiotestit"
+    runs-on: ubuntu-latest
```

Asetetaan integraatiotesteille timeout, joka on lyhyempi kuin Github Actionsin default timeout.
Näiden testien suhteen on riskinä, että ActiveMQ yhteyden hajotessa testit rullaa monta tuntia. Tämä puolestaan voi aiheuttaa ylimääräistä laskutusta.

Integraatiotestit tuskin koskaan kestää puolta tuntia pidempään, joten laitetaan timeoutiksi vaikka 30 minuuttia.
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes

```suggestion
    runs-on: ubuntu-latest
    timeout-minutes: 30
```